src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -301,17 +374,113 @@ public void setBaitSetName(final String name) {
             this.metrics.PROBE_SET = name;
         }
 
+        /** Get the the number of bases in the given alignment block and record that have base quality greater or equal to the minimum */
+        private int getNumBasesPassingMinimumBaseQuality(final SAMRecord record, final AlignmentBlock block) {
+            int basesInBlockAtMinimumQuality = 0;
+            final byte[] baseQualities = record.getBaseQualities();
+            for (int idx = block.getReadStart(); idx <= CoordMath.getEnd(block.getLength(), block.getReadStart()); idx++) { // idx is one-based
+                if (minimumBaseQuality <= baseQualities[idx-1]) basesInBlockAtMinimumQuality++;
+            }
+            return basesInBlockAtMinimumQuality;
+        }
+
+        private int getNumOverlappingBasesToClip(final SAMRecord rec) {
+            // NB: ignores how to handle supplemental records when present for both ends by just using the mate information in the record.
+
+            if (!clipOverlappingReads || !rec.getReadPairedFlag()) return 0;
+
+            // Only clip records that are left-most in genomic order and overlapping.
+            if (rec.getMateAlignmentStart() < rec.getAlignmentStart()) return 0; // right-most, so ignore.  Do this first since getAlignmentEnd is expensive
+
+            final int numBasesToClip = rec.getAlignmentEnd() - rec.getMateAlignmentStart()  + 1;
+            if (numBasesToClip < 0) return 0; // left-most but not overlapping
+
+            return numBasesToClip;
+        }
+
+        private SAMRecord clipOverlappingBases(final SAMRecord rec, final int numOverlappingBasesToClip) {
+            // NB: ignores how to handle supplemental records when present for both ends by just using the mate information in the record.
+
+            if (!clipOverlappingReads || numOverlappingBasesToClip <= 0) return rec;
+
+            try {
+                // watch out for when the second read overlaps all of the first read
+                if (rec.getMateAlignmentStart() <= rec.getAlignmentStart()) { // make it unmapped
+                    if (noSideEffects) {
```

Comment:
factor out the cloning logic to the beginning of this function to avoid much code duplication below